net/tailscale/files/tailscale.init

```diff
@@ -4,44 +4,258 @@
 # Copyright (C) 2021 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 # SPDX-License-Identifier: Apache-2.0
 
+START=90
+
 USE_PROCD=1
-START=80
 
-start_service() {
-  local state_file
-  local port
-  local std_err std_out
+PROG=/usr/sbin/tailscale
+PROGD=/usr/sbin/tailscaled
+CONFIG_PATH=/var/lib/tailscale
+
+service_triggers() {
+	procd_add_reload_trigger "tailscale"
+	procd_add_interface_trigger "interface.*.up" wan /etc/init.d/tailscale reload
+}
+
+section_enabled() {
+	config_get_bool enabled "$1" 'enabled' 0
+	[ $enabled -gt 0 ]
+}
+
+custom_instance() {
+	local cfg="$1"
+	local accept_routes hostname accept_dns advertise_exit_node exit_node advertise_routes disable_snat_subnet_routes subnet_routes flags login_server authkey std_out std_err
+
+	config_get_bool accept_routes $cfg 'accept_routes'
+	config_get hostname $cfg 'hostname'
+	config_get_bool accept_dns $cfg 'accept_dns'
+	config_get_bool advertise_exit_node $cfg 'advertise_exit_node'
+	config_get exit_node $cfg 'exit_node'
+	config_get advertise_routes $cfg 'advertise_routes'
+	config_get_bool disable_snat_subnet_routes $cfg 'disable_snat_subnet_routes'
+	config_get flags $cfg 'flags'
+	config_get login_server $cfg 'login_server'
+	config_get authkey $cfg 'authkey'
+	config_get_bool std_out $cfg 'log_stdout'
+	config_get_bool std_err $cfg 'log_stderr'
+
+	procd_open_instance
+	procd_set_param command $PROG up --reset
+
+	[ "$accept_routes" = "1" ] && procd_append_param command --accept-routes=true
+	[ -n "$hostname" ] && procd_append_param command --hostname="$hostname"
+	[ "$accept_dns" = "0" ] && procd_append_param command --accept-dns=false
+	[ "$advertise_exit_node" = "1" ] && procd_append_param command --advertise-exit-node
+	[ -n "$exit_node" ] && {
+		procd_append_param command --exit-node="$exit_node"
+		procd_append_param command --exit-node-allow-lan-access=true
+	}
+	[ -n "$advertise_routes" ] && procd_append_param command --advertise-routes="$(echo $advertise_routes | tr ' ' ',')"
+	[ "$disable_snat_subnet_routes" = "1" ] && procd_append_param command --snat-subnet-routes=false
+	[ -n "$flags" ] && procd_append_param command $flags
+	[ -n "$login_server" ] && procd_append_param command --login-server="$login_server"
+	[ -n "$authkey" ] && procd_append_param command --authkey="$authkey"
+
+	procd_set_param stdout "$std_out"
+	procd_set_param stderr "$std_err"
+	procd_close_instance
+	(
+		# Use flock to acquire an exclusive lock
+		local LOCK_FILE=/var/lock/tailscale.lock
+		trap 'rm -f "$LOCK_FILE"; exit' INT TERM EXIT
```

Comment:
This will call the trap when failing to acquire the lock (while another instance is running), so when this code is executed again there will be no lock file and it can be re-entered, even if the original instance is still running.

So the trap must be defined _after_ locking.